src/javascript/app/common/data_manager.js

```diff
@@ -0,0 +1,102 @@
+import {
+    triggerBarrierChange,
+    triggerContractChange,
+    triggerPurchaseChange,
+    triggerTimeChange,
+    triggerTradeChange,
+} from '../hooks/events';
+
+class DataManager {
+    constructor() {
+        this.data = {
+            trade   : {},
+            purchase: {},
+            contract: {},
+        };
+    }
+
+    set(data, data_type, optional) {
```

Comment:
it is not clear what the `optional` prop does

Reply:
I changed the parameter name to optional_trigger. this is an optional param to set function to triggerBarrierChange,  triggerTimeChange only and to separate from tradeChange for not re-rendering everytime.